apps/ui/src/components/layouts/LanguageLayout.tsx

```diff
@@ -17,25 +17,47 @@
  * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-export const formatDate = (date: Date): string => {
-	const monthNames = [
-		'January',
-		'February',
-		'March',
-		'April',
-		'May',
-		'June',
-		'July',
-		'August',
-		'September',
-		'October',
-		'November',
-		'December',
-	];
-
-	const day = date.getDate().toString().padStart(2, '0');
-	const month = monthNames[date.getMonth()]?.substring(0, 3);
-	const year = date.getFullYear();
-
-	return `${month} ${day}, ${year}`;
+import { Layout } from 'antd';
+import { useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Outlet, useLocation, useNavigate, useParams } from 'react-router';
+
+import HeaderComponent from '@/components//Header';
+import FooterComponent from '@/components/Footer';
+import { resources } from '@/i18n/translations';
+
+const LanguageLayout = () => {
+	const { lang } = useParams();
+	const { i18n } = useTranslation();
+	const { pathname } = useLocation();
+	const navigate = useNavigate();
+
+	useEffect(() => {
+		const languages = Object.keys(resources);
+
+		// Continue as default language
+		if (!lang) {
+			return;
+		}
+
+		// If the language is not identified, continue as default but remove the param
+		if (!languages.includes(lang)) {
+			const cleanedUrl = pathname.replace(`${lang}/`, '');
```

Comment:
I'm wondering if we should figure out what the actual behaviour for this should be. 

Looking at other provincial and federal bilingual sites specifically when no language identifier is specified, the website defaults to a language selection screen (see: https://www.canada.ca/ and https://www.ontario.ca/), whereas, if a unknown language is specified (e.g https://www.canada.ca/es/ or https://www.ontario.ca/es/), it results in a 404. 

I'm wondering if it'd be wise to not assume english as our "default", but rather go to a language select splash page like Canada.ca / Ontario.ca does?

Reply:
Side note: these websites also translate the path URL (eg. https://www.ontario.ca/fr/page/gouvernement-de-lontario vs. https://www.ontario.ca/page/government-ontario), but this might be out of scope for now

Reply:
These are definitely good suggestions worth considering. I think we'd have to involve Alexis and Patrick in designing what that looks like, and draft up a new ticket. Right now there's no design for a 'select language screen' so it's a bit vague how James would implement one.

We could potentially look at, no language = default to english, unrecognized language = throw error, for the scope of this PR?

Translating path urls as well I think there's good arguments for adding, but again this expands the scope of the ticket and might be worth discussing with the broader team.

Reply:
> We could potentially look at, no language = default to english, unrecognized language = throw error, for the scope of this PR?

Initially, this was my idea, recongnize `en` and `fr` then default to english if it doesn't register. But I was having a problem with how [optional segments](https://reactrouter.com/start/library/routing#optional-segments) works with the current solution. So right now, if the language is not identified, it removes the optional param via the current logic. But it only really works if the path is not at the root.

For example: `www.example.com/emg/dashboard` will translation to `www.example.com/dashboard` and default to english, but `www.example.com/emg` will persist in the url but will still render the homepage as english (Since the url is optional parameter works at the highest level in the url and accepts anything). So adding a error page is not really possible with this solution. 

I sat on this for quite sometime and didn't want to linger too long as theres many different ways to handle this. I figured a discussion for a temporary language switch solution would be more appropriate before wasting too much time.

Also thank you @surajgoraya for bringing this up, I wanted to talk about this yesterday but I was out and I missed it in my review!

Reply:
I think `www.example.com/emg/dashboard` should reroute to `www.example.com/en/dashboard` and `www.example.com/emg` should route to a 404 page, would that be possible?

Or, we treat anything following root that is not recognized as a 404, so both `www.example.com/emg/dashboard` and `www.example.com/emg`
So make language required

Reply:
Discussed offline, I'll let James fill in